azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentApiLiveTest.java

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import org.jclouds.azurecompute.arm.domain.Deployment;
+import org.jclouds.azurecompute.arm.domain.Deployment.ProvisioningState;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "DeploymentApiLiveTest", singleThreaded = true)
+public class DeploymentApiLiveTest extends BaseAzureComputeApiLiveTest {
+    private int maxTestDuration = 90;
+    private int pollingInterval = 3; // how frequently to poll for create status
+
+    private String resourceGroup;
+    private String resourceName;
+    private String deploymentName;
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+        resourceGroup = getResourceGroupName();
+    }
+
+    private String getPutBody(String template, String mode, String parameters) {
+        String body = "{ " +
+                      "\"properties\" : " +
+                      "  { " +
+                      "    \"template\" : " + template + ", " +
+                      "    \"mode\" : \"" + mode + "\", " +
+                      "    \"parameters\" : " + parameters + " " +
+                      "  } " +
+                      "}";
+        return body;
+    }
+
+    @Test(groups = "live")
+    public void testCreate() {
+        Long now = System.currentTimeMillis();
+        resourceName = "jcres" + now;
+        deploymentName = "jcdep" + now;
+        String template = "{\"$schema\":\"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\"contentVersion\":\"1.0.0.0\",\"parameters\":{\"newStorageAccountName\":{\"type\":\"string\",\"metadata\":{\"description\":\"Name of the Storage Account\"}},\"storageAccountType\":{\"type\":\"string\",\"defaultValue\":\"Standard_LRS\",\"allowedValues\":[\"Standard_LRS\",\"Standard_GRS\",\"Standard_ZRS\"],\"metadata\":{\"description\":\"Storage Account type\"}},\"location\":{\"type\":\"string\",\"allowedValues\":[\"East US\",\"West US\",\"West Europe\",\"East Asia\",\"Southeast Asia\"],\"metadata\":{\"description\":\"Location of storage account\"}}},\"resources\":[{\"type\":\"Microsoft.Storage/storageAccounts\",\"name\":\"[parameters('newStorageAccountName')]\",\"apiVersion\":\"2015-05-01-preview\",\"location\":\"[parameters('location')]\",\"properties\":{\"accountType\":\"[parameters('storageAccountType')]\"}}]}";
+        String parameters = "{\"newStorageAccountName\":{\"value\":\"" + resourceName + "\"},\"storageAccountType\":{\"value\":\"Standard_LRS\"},\"location\":{\"value\":\"West US\"}}";
+        String properties = getPutBody(template, "Incremental", parameters);
```

Comment:
>The plan was to basically reflect the raw API itself, thus the three JSON inputs

Well, the three inputs only exist in the test. The method int he API just gets one raw JSON parameter, which turns to be the request body. Every single method in the raw API accepts a json payload. In this case, it happens to be a deployment template, but all other method in the API accept raw jsons too, be it a network, a disk or a deployment template. Why should we be providing high level and object oriented methods for some methods, and some other methods that require already serialized json? I understand that the template is a json object, but I think the user experience is not good here: users need to know serialization details. Given that we already have builder objects to be the template "users should be serializing", couldn't we let the api method accept what that builder produces, and let jclouds serialize it? Looks much less error-prone and a better user experience?

>Would users ever call the DeploymentApi directly

jclouds should allow users to call the DeploymentApi directly, yes. And they'll do, eventually, when they want to tweak a deployment in a way the portable abstraction does not let them to do so.